packages/js-sdk/src/sandbox/index.ts

```diff
@@ -344,26 +367,77 @@ export class Sandbox extends SandboxApi {
    *
    * @param path the directory where to upload the file, defaults to user's home directory.
    *
+   * @param useSignature URL will be signed with the access token, this can be used for uploading files to the sandbox from a different environment (e.g. browser).
+   *
+   * @param signatureExpirationInSeconds URL will be signed with the access token, this can be used for uploading files to the sandbox from a different environment (e.g. browser).
+   *
    * @returns URL for uploading file.
    */
-  uploadUrl(path?: string) {
-    return this.fileUrl(path)
+  uploadUrl(path?: string, useSignature?: boolean, signatureExpirationInSeconds?: number) {
+    if (!this.envdAccessToken && (useSignature != undefined || signatureExpirationInSeconds != undefined)) {
+      throw new Error('Signature can be used only when sandbox is spawned with secure option.')
+    }
+
+    if (useSignature == undefined && signatureExpirationInSeconds != undefined) {
+      throw new Error('Signature expiration can be used only when signature is set to true.')
+    }
+
+   const fileUrl = this.fileUrl(path, defaultUsername)
```

Comment:
i know we don't have formatting set up, but at least keep the same spacing 🙏🏻 
```suggestion
    const fileUrl = this.fileUrl(path, defaultUsername)
```